### ISSUE TITLE
The experimentalObjectRestSpread feature does not work with complex code

### ISSUE THREAD

**Issue by gre** [eslint/eslint#3030]
version: 1.0.0-rc-1

```js

const foo = {
  a: 1
}
const obj = {
  foo
};
const bar = "bar";
const res = {
  ...foo,
  bar
};

// This is supposed to work too:
const res2 = {
  ...obj.foo,
  bar
};
```

<img width="411" alt="screen shot 2015-07-16 at 16 17 10" src="https://cloud.githubusercontent.com/assets/211411/8725732/22e947f0-2bd6-11e5-9e46-b94635af82fe.png">

```
eslint test.js

test.js
  16:10  error  Unexpected token

✖ 1 problem (1 error, 0 warnings)
```

**Comment by gyandeeps** [comment]
What is the version of eslint are you using?
Can share the console output of eslint?

**Comment by xjamundx** [comment]
Considering this merged only 22hours ago https://github.com/eslint/espree/issues/163

I assume it's not been published to eslint yet.

**Comment by gyandeeps** [comment]
@xjamundx Its on the our 1.0.0-rc release candidate

**Comment by xjamundx** [comment]
Oh wow. Things move fast. Go on vacation and the world moves along.

**Comment by IanVS** [comment]
You underestimate the speed of @nzakas. :)

**Comment by gre** [comment]
I have updated the initial post about version and output :)

**Comment by michaelficarra** [comment]
Yes, the proposal's grammar allows AssignmentExpression in object spread position, so it looks like espree is broken. But you should open your issue there, not here.

**Comment by gre** [comment]
thanks @michaelficarra

**Comment by michaelficarra** [comment]
@gre You can close this issue now.
